lightning/src/onion_message/messenger.rs

```diff
@@ -243,6 +243,54 @@ impl OnionMessageRecipient {
 	}
 }
 
+pub struct Responder<'a, OMH: OnionMessageHandler, T: OnionMessageContents> {
+	messenger: &'a OMH,
+	pub message: T,
```

Comment:
Let's leave the `message` out of responder and put it directly in `ReceivedOnionMessage::WithReplyPath`. You can add a `&'static str` field for the message type.

Reply:
Sounds reasonable! Addressed in [8526afc](https://github.com/lightningdevkit/rust-lightning/pull/2907/commits/8526afc0989c69249a4fca1f10d03d15dc1a9c15)

Also instead of introducing message type as a separate variable, I used the fact that we enforce response and message to be of the same type at compile time.
Let me know if we should rather have a separate message type variable, and I would love to make that change.